android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletResetActivity.java

```diff
@@ -0,0 +1,163 @@
+
+        DialogInterface.OnClickListener onClickListener = new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int button) {
+                if (button == AlertDialog.BUTTON_POSITIVE) {
+                    if (mKeyringController != null) {
+                        String inputText = input.getText().toString().trim();
+                        if (TextUtils.equals(inputText, "Yes")) {
+                            Log.w(TAG, "Reset");
+                            mKeyringController.reset();
+                        }
+                    } else {
+                        Log.w(TAG, "mKeyringController == null");
+                    }
+                } else {
+                    dialog.dismiss();
+                }
+            }
+        };
+
+        AlertDialog.Builder alert =
+                new AlertDialog.Builder(this, R.style.Theme_Chromium_AlertDialog);
+        mAlertDialog =
+                alert.setTitle(R.string.brave_wallet_reset_settings_option)
+                        .setView(view)
+                        .setPositiveButton(
+                                R.string.brave_wallet_confirm_text, onClickListener)
+                        .setNegativeButton(R.string.cancel, onClickListener)
+                        .create();
+        mAlertDialog.getDelegate().setHandleNativeActionModesEnabled(false);
+        mAlertDialog.setOnShowListener(new DialogInterface.OnShowListener() {
+            @Override
+            public void onShow(DialogInterface dialog) {
+                mOkButton = ((AlertDialog) dialog).getButton(AlertDialog.BUTTON_POSITIVE);
+                mOkButton.setEnabled(false);
+                KeyboardVisibilityDelegate.getInstance().showKeyboard(input);
+            }
+        });
+        mAlertDialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
+            @Override
+            public void onDismiss(DialogInterface dialog) {
+                finish();
+            }
+        });
+
+        input.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void afterTextChanged(Editable s) {}
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                // Disable ok button if input is invalid
+                String inputText = s.toString().trim();
+
+                if (mOkButton != null) {
+                    mOkButton.setEnabled(TextUtils.equals(inputText, "Yes"));
+                }
+            }
+        });
+
+        onInitialLayoutInflationComplete();
+    }
+
+    @Override
+    public void onResume()
+    {
+        super.onResume();
+
+        mAlertDialog.show();
+    }
+
+    @Override
+    public void onConnectionError(MojoException e) {
+        mKeyringController = null;
```

Comment:
I also find out that mojo calls required a call `mKeyringController.close()` it says that there is a leak possible if you don't call. So please call it here and also `onDestroy` for that Activity.